amxmodx/CoreConfig.cpp

```diff
@@ -0,0 +1,324 @@
+	m_LegacyMainConfigExecuted = false;
+	m_LegacyMapConfigsExecuted = false,
+	m_legacyMapConfigNextTime = 0.0f;
+}
+
+void CoreConfig::ExecuteMainConfig()
+{
+	if (m_LegacyMainConfigExecuted)
+	{
+		return;
+	}
+
+	char path[PLATFORM_MAX_PATH];
+	char command[PLATFORM_MAX_PATH + sizeof(CommandFormat)];
+
+	UTIL_Format(path, sizeof(path), "%s/%s/%s", g_mod_name.chars(), get_localinfo("amx_configdir", "addons/amxmodx/configs"), MainConfigFile);
+	UTIL_Format(command, sizeof(command), CommandFormat, path);
+
+	SERVER_COMMAND(command);
+}
+
+void CoreConfig::ExecuteAutoConfigs()
+{
+	for (size_t i = 0; i < static_cast<size_t>(g_plugins.getPluginsNum()); ++i)
+	{
+		auto plugin = g_plugins.findPlugin(i);
+
+		bool can_create = true;
+
+		for (size_t j = 0; j < plugin->GetConfigCount(); ++j)
+		{
+			can_create = ExecuteAutoConfig(plugin, plugin->GetConfig(j), can_create);
+		}
+	}
+
+	executeForwards(m_ConfigsBufferedForward);
+}
+
+bool CoreConfig::ExecuteAutoConfig(CPluginMngr::CPlugin *plugin, AutoConfig *config, bool can_create)
+{
+	bool will_create = false;
+
+	const char *configsDir = get_localinfo("amx_configdir", "addons/amxmodx/configs");
+
+	if (can_create && config->create)
+	{
+		will_create = true;
+
+		const char *folder = config->folder.chars();
+
+		char path[PLATFORM_MAX_PATH];
+		char build[PLATFORM_MAX_PATH];
+
+		build_pathname_r(path, sizeof(path), "%s%s/%s", configsDir, AutoConfigDir, folder);
+
+		if (!g_LibSys.IsPathDirectory(path))
+		{
+			char *cur_ptr = path;
+
+			g_LibSys.PathFormat(path, sizeof(path), "%s", folder);
+			build_pathname_r(build, sizeof(build), "%s%s", configsDir, AutoConfigDir);
+
+			size_t length = strlen(build);
+
+			do
+			{
+				char *next_ptr = cur_ptr;
+
+				while (*next_ptr != '\0')
+				{
+					if (*next_ptr == PLATFORM_SEP_CHAR)
+					{
+						*next_ptr = '\0';
+						next_ptr++;
+						break;
+					}
+
+					next_ptr++;
+				}
+
+				if (*next_ptr == '\0')
+				{
+					next_ptr = nullptr;
+				}
+
+				length += g_LibSys.PathFormat(&build[length], sizeof(build) - length, "/%s", cur_ptr);
+
+				if (!g_LibSys.CreateFolder(build))
+				{
+					break;
+				}
+
+				cur_ptr = next_ptr;
+
+			} while (cur_ptr);
+		}
+	}
+
+	char file[PLATFORM_MAX_PATH];
+
+	if (config->folder.length())
+	{
+		UTIL_Format(file, sizeof(file), "%s/%s%s/%s/%s.cfg", g_mod_name.chars(), configsDir, AutoConfigDir, config->folder.chars(), config->autocfg.chars());
+	}
+	else
+	{
+		UTIL_Format(file, sizeof(file), "%s/%s%s/%s.cfg", g_mod_name.chars(), configsDir, AutoConfigDir, config->autocfg.chars());
+	}
+
+	bool file_exists = g_LibSys.IsPathFile(file);
+
+	if (!file_exists && will_create)
+	{
+		auto list = g_CvarManager.GetCvarsList();
+
+		if (list->empty())
+		{
+			return can_create;
+		}
+
+		FILE *fp = fopen(file, "wt");
+
+		if (fp)
+		{
+			fprintf(fp, "// This file was auto-generated by AMX Mod X (v%s)\n", AMXX_VERSION);
+			fprintf(fp, "// Cvars for plugin \"%s\"\n", plugin->getName());
```

Comment:
How about using plugin title and author (and perhaps additionally version it was generated for)?